opam-ci-check/lib/lint.mli

```diff
@@ -5,34 +5,55 @@
 include module type of Lint_error
 
 module Checks : sig
-  val package_name_collision: string -> string -> bool
+  val package_name_collision : string -> string -> bool
   (** [package_name_collision p0 p1] returns true if [p0] is similar to [p1].
     Similarity is defined to be:
 
     - Case-insensitive string equality considering underscores ([_])
       dashes ([-]), and the empty string to be equal *)
 end
 
-val check :
-  new_pkgs:string list ->
-  changed_pkgs:string list ->
-  string ->
+val get_packages : string -> string list
+(** [get_packages repo] is a list of all the packages in the opam repository
+    located at [repo].
+
+    Example:
+
+    {[
+      let all_packages = get_packages repo
+    ]} *)
+
+val make_parse_error : OpamPackage.t -> OpamPackage.t * error
```

Comment:
With the new API, which expects the opam file to already be parsed, I don't think it makes sense for the parse error to be part of the lint API... so, IMO, this suggest a choice:

1. Either accept that parsing opam files is out of scope for the CI linting logic, and move that into (e.g.) the analysis stage.
2. Keep parsing as part of the the linting.

Support for (1) may be drawn from considering that we are already parsing all opam files ar part of the analysis phase (https://github.com/ocurrent/opam-repo-ci/blob/ec9e8ae658a0b6363db848db4b1d8d0082db034d/lib/analyse.ml#L142-L148), and noting that https://github.com/ocaml/opam-repository/issues/26956 indicates we may want opam parsing failures to be more visible category of failure than just a linting error.

Support for (2) may be found in the fact that parsing errors have been parting of linting up to this point, and that we generally expect parse failures to be reported in our linting tools.

I suggest we follow (2), even if we ultimately want to consider (1), because it is a smaller change. To achieve (2), we could tweak the new API so that `lint_package` takes a `string` as its principle argument, and then parse the string internally. If we want to support both a string and an already parsed opam file, we could have the principle argument be something like

```
[`Parsed of OpamFile.OPAM.t | `Raw of string]
```

It is possible I am being to pedantic, but with long term viability in view, I think it is not a great idea to have a package like this holding a data type for a kind of data it cannot produce.

WDYT?

Reply:
I agree that the parse_error no longer should be a part of the API. And we can rely on the analysis phase catching any broken opam files.  Currently, we shouldn't even be triggering the lint step in cases where the analysis fails. 

For, the CLI also we can fail hard instead of trying to continue when one of the files fails to parse. I don't see a use case for when we'd want to allow one package with a parse error while parsing other files. I think it will also tie in with the other suggestion to remove `get_packages` from the public API and provide a bulk processing API.

Reply:
I've refactored this to completely get rid of the ParseError since we don't need to be "well behaved" when an opam file fails to parse, and can fail hard.